scripts/lint-style.py

```diff
@@ -149,6 +152,45 @@ def long_lines_check(lines, path):
             errors += [(ERR_LIN, line_nr, path)]
     return errors
 
+def indent_check(lines, path):
+    errors = []
+    indent_lvl = 0
+    in_prf = 0  # counter for nested proof blocks
+    check_rest_of_block = True  # we only check uncomplicated syntax
+    ended_with_comma = False  # track whether the previous line ends with a comma
+    inside_special = 0  # track whether we are inside ⟨⟩ or []
+    for line_nr, line in enumerate(lines, 1):
+        lstr = line.lstrip(' ')  # strip spaces from beginning of line
+        if in_prf > 0 and check_rest_of_block and ended_with_comma and not inside_special:
+            if lstr[0] == '{' and len(line) - len(lstr) != indent_lvl:
+                errors += [(WRN_IND, line_nr, path)]
+        ended_with_comma = (line.endswith(",\n"))
+        inside_special += line.count('⟨') + line.count('[') - line.count('⟩') - line.count(']')
+        if line[0] != ' ':
+            # reset the state
+            indent_lvl = 0
+            in_prf = 0
+            check_rest_of_block = True
+            ended_with_comma = False
+            inside_special = 0
+        if "match" in line or "calc" in line:
```

Comment:
I'm concerned checks like `if "keyword" in line` are too likely to get false positives, e.g. the check for `end` will match all lines containing `linear_indep`**`end`**`ent`, and cause the `in_prf` counter to turn negative. (It's good that you explicitly say `in_prf > 0` above rather than `in_prf`, or we'd get some weird errors!)

More robust would be to use `re.match("\bkeyword\b", line)` (where `\b` is regex syntax for "word boundary").

Reply:
Fixed in the other PR